helpers/any.js

```diff
@@ -15,25 +14,47 @@ const factory = () => {
         const opts = args.pop();
         const predicate = opts.hash;
 
-        if (!_.isEmpty(predicate)) {
+        if (!common.isEmptyObject(predicate)) {
             // With options hash, we check the contents of first argument
-            any = _.any(args[0], predicate);
+            const predicateKeys = Object.keys(predicate);
+            const arg = args[0];
+
+            function matchesPredicate(element, predicate) {
+                for (let i = 0; i < predicateKeys.length; i++) {
+                    if (!Object.is(element[predicateKeys[i]], predicate[predicateKeys[i]])) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+
+            if (Array.isArray(arg)) {
+                for (let i = 0; i < arg.length; i++) {
+                    const element = arg[i];
+                    any = matchesPredicate(element, predicate);
+                    if (any) {
+                        break;
+                    } 
+                }
+            } else if (common.isObject(arg)) {
+                const argIndices = Object.keys(arg);
+                for (let i = 0; i < argIndices.length; i++) {
```

Comment:
I don't think this is the behavior we want. When the first argument is an object, we should be testing that the object has the same properties/values defined in the options hash.

This code appears to be doing that for each attribute of the input object (i.e. assuming a nested object).

Instead, I think you would just do
```
else if (common.isObject(arg)) {
  any = matchesPredicate(arg, predicate);
}
```

Reply:
I tested the test suite with _.any just now and it matches the current code behavior.